### PR TITLE
chore: update workflows from templates

### PR DESCRIPTION
**Pull request by skjnldsv**
This PR was automatically generated by the workflow dispatcher
Please check that there are no reverted changes before merging ⚠

See templates in https://github.com/nextcloud/.github

### CODE REVIEW

**Review by st3iny — APPROVED** [review]
This should fix static-analysis.

**Comment by st3iny** [comment]
@ChristophWurst Please have a look at the changes. Especially at the changes in composer.json (dropped packages). I did some testing and it continued to work fine.

File: composer.json [hunk 1]
```diff
@@ -15,11 +15,9 @@
 		"php": ">=7.4.0",
 		"ext-json": "*",
 		"bamarni/composer-bin-plugin": "^1.5",
-		"christophwurst/nextcloud-http-client": "^0.2.0",
```
Comment (ChristophWurst) [inline composer.json:18]: I need to check this. The httplug package *detects* the HTTP client dynamically. I'm curious which implementation it depects without this meta package.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ jobs:
     name: php${{ matrix.php-version }}-${{ matrix.db }} integration tests
     services:
       mysql-service:
-        image: mariadb:10.5
+        image: ghcr.io/nextcloud/continuous-integration-mariadb-10.6:latest
```
[the remark this answers is not in the document]
Comment (st3iny) [reply .github/workflows/test.yml:68]: This workaround was suggested via an internal mail from John (June 1st).

[the remark this answers is not in the document]
Comment (st3iny) [reply .github/workflows/test.yml:68]: Ref https://github.com/nextcloud/docker-ci

[the remark this answers is not in the document]
Comment (st3iny) [reply .github/workflows/test.yml:68]: Yeah, cron jobs for the most important ones would be great. Maybe build them once a week. Oh and oracle db should also be included.

File: composer.json (hunk 1 above)
Comment (ChristophWurst) [reply composer.json:18]: I don't remember why this was added. Right now https://packagist.org/packages/web-auth/webauthn-lib doesn't depend on any of the virtual packages provided by https://packagist.org/packages/christophwurst/nextcloud-http-client.

Comment (st3iny) [reply composer.json:18]: I did another test (login, personal settings and setup) and it works fine.

**Pull request merged**